applications/main/infrared/infrared_cli.c

```diff
@@ -333,57 +340,140 @@ static void infrared_cli_process_decode(Cli* cli, FuriString* args) {
+            printf("Must supply a valid signal for type of remote selected.\r\n");
+            break;
+        }
 
-            uint32_t record_count;
-            uint32_t index = 0;
+        infrared_cli_brute_force_signals(cli, Remote, signal);
+        break;
 
-            bool running = infrared_brute_force_start(brute_force, index, &record_count);
+    } while(false);
 
-            while(running) {
-                running = infrared_brute_force_send_next(brute_force);
-            }
+    furi_string_free(command);
+    furi_string_free(remote);
+    furi_string_free(signal);
+}
+
+static void infrared_cli_list_remote_signals(enum RemoteTypes remote_type) {
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    FlipperFormat* ff = flipper_format_buffered_file_alloc(storage);
+    const char* remote_file = NULL;
+    bool success = false;
+
+    switch(remote_type) {
+    case TV:
+        remote_file = EXT_PATH("infrared/assets/tv.ir");
+        break;
+    case AC:
+        remote_file = EXT_PATH("infrared/assets/ac.ir");
+        break;
+    default:
+        break;
+    }
+
+    success = flipper_format_buffered_file_open_existing(ff, remote_file);
+    if(success) {
+        FuriString* signal_name;
+        signal_name = furi_string_alloc();
+        printf("Valid signals:\r\n");
+        while(flipper_format_read_string(ff, "name", signal_name)) {
+            printf("\t%s\r\n", furi_string_get_cstr(signal_name));
+        }
+        furi_string_free(signal_name);
+    }
+
+    flipper_format_free(ff);
+    furi_record_close(RECORD_STORAGE);
+}
 
-            infrared_brute_force_stop(brute_force);
+static void
+    infrared_cli_brute_force_signals(Cli* cli, enum RemoteTypes remote_type, FuriString* signal) {
+    InfraredBruteForce* brute_force = infrared_brute_force_alloc();
+    const char* remote_file = NULL;
+    uint32_t i = 0;
+
+    switch(remote_type) {
+    case TV:
+        remote_file = EXT_PATH("infrared/assets/tv.ir");
+        break;
+    case AC:
+        remote_file = EXT_PATH("infrared/assets/ac.ir");
+        break;
+    default:
+        break;
+    }
+
+    infrared_brute_force_set_db_filename(brute_force, remote_file);
+    infrared_brute_force_add_record(brute_force, i++, furi_string_get_cstr(signal));
+
+    bool success = infrared_brute_force_calculate_messages(brute_force);
+    if(success) {
+        uint32_t record_count;
+        uint32_t index = 0;
+        int records_sent = 0;
+
+        bool running = infrared_brute_force_start(brute_force, index, &record_count);
+
+        printf("Sending %ld codes to the tv.\r\n", record_count);
+        printf("Press Ctrl-C to stop.\r\n");
+        while(running) {
+            running = infrared_brute_force_send_next(brute_force);
+            records_sent++;
+
+            if(cli_cmd_interrupt_received(cli)) break;
+
+            int pc = (float)records_sent / (float)record_count * 100;
+            printf("\r%d%% complete.", pc);
```

Comment:
As of now, prints "125%" at finish, which is incorrect.

Reply:
Thanks. Looking  into!